sky/provision/aws/instance.py

```diff
@@ -718,15 +718,19 @@ def open_ports(
     existing_ports: Set[int] = set()
     for existing_rule in sg.ip_permissions:
         # Skip any non-tcp rules.
-        if existing_rule['IpProtocol'] != 'tcp':
+        if existing_rule['IpProtocol'] not in ['tcp', '-1']:
             continue
         # Skip any rules that don't have a FromPort or ToPort.
-        if 'FromPort' not in existing_rule or 'ToPort' not in existing_rule:
-            continue
-        existing_ports.update(
-            range(existing_rule['FromPort'], existing_rule['ToPort'] + 1))
-    ports_to_open = resources_utils.port_set_to_ranges(
-        resources_utils.port_ranges_to_set(ports) - existing_ports)
+        if 'FromPort' in existing_rule and 'ToPort' in existing_rule:
+            existing_ports.update(
+                range(existing_rule['FromPort'], existing_rule['ToPort'] + 1))
+        elif existing_rule['IpProtocol'] == '-1':
+            existing_ports.add(-1)
```

Comment:
If I'm understanding your question correctly, since AWS SG rules are permissive only, no deny rule can pre-exist in the security group. Or are you asking about a case where a deny rule is provided within Skypilot?

Reply:
added a TODO in case a Skypilot deny scenario were to exist.

Reply:
I see! That makes sense. Thanks for elaborating this. In that case, the TODO message must not be needed. : )